p3-frontend/src/environments/environment.prod.ts

```diff
@@ -1,3 +1,9 @@
 export const environment = {
   production: true
 };
+
+export const REGISTRY_URL = `http://gateway:8085/api/`;
+
+export const SessionVar = {
+  loginKey : "userLogin"
+}
```

Comment:
yes, the prod environment file needs to define everything the other one does because it replaces the other one in the prod build, which i guess the last iteration neglected.

what i'd suggest is leave the environment.ts with the configuration most useful for the devs, which may be localhost on one or another port depending on if the backend is via docker-compose or via some spring/eclipse IDE. (ideally, you'd set the compose file to use the same ports that the IDE will; that way, environment.ts can work for both without edits.)

environment.prod.ts, though, should work in the cluster. in this case, depending on the ingress, you can use something like `` `${location.origin}/api/` ``. location is a global variable in javascript with some other properties; location.origin will ask the browser to fill in whatever domain it's currently looking at the webpage from. so it'll automatically be the right domain.
(somewhat unrelated side note, you can run an angular app in docker for local testing, but there's not a whole lot of reason to in practice compared to backend code.)

Reply:
Isaac is planning to integrate this into his plan for the ingress solution for the prod side.

I switched back to localhost on the non-prod side per your suggestion.  It is now in the initial state the former dev team set.